webapp/src/components/CreateEstatePage/EstateSelect/EstateSelect.js

```diff
@@ -0,0 +1,134 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { Container } from 'semantic-ui-react'
+
+import ParcelPreview from 'components/ParcelPreview'
+import EstateSelectActions from './EstateSelectActions'
+import Parcel from 'components/Parcel'
+import { t } from 'modules/translation/utils'
+import { isOwner } from 'modules/parcels/utils'
+import { coordsType } from 'components/types'
+
+import './EstateSelect.css'
+
+export default class EstateSelect extends React.PureComponent {
+  static propTypes = {
+    x: PropTypes.number.isRequired,
+    y: PropTypes.number.isRequired,
+    error: PropTypes.string,
+    onCancel: PropTypes.func.isRequired,
+    onContinue: PropTypes.func.isRequired,
+    onChange: PropTypes.func.isRequired,
+    value: PropTypes.arrayOf(coordsType).isRequired
+  }
+
+  isNeighbour = (x, y) => coords => {
+    return (
+      (coords.x === x && (coords.y + 1 === y || coords.y - 1 === y)) ||
+      (coords.y === y && (coords.x + 1 === x || coords.x - 1 === x))
+    )
+  }
+
+  hasNeighbour = (x, y) => {
+    const { value: parcels } = this.props
+    return parcels.some(this.isNeighbour(x, y))
+  }
+
+  getNeighbours = (x, y, parcels = this.props.parcels) => {
+    return parcels.filter(this.isNeighbour(x, y))
+  }
+
+  areConnected = (parcels, alreadyTraveled = [], remaining = [...parcels]) => {
+    if (alreadyTraveled.length === parcels.length) {
+      return true
+    }
+
+    if (remaining.length === 0) {
+      return false
+    }
+
+    let actual = remaining.pop()
+
+    const neighbours = this.getNeighbours(actual.x, actual.y, parcels).filter(
+      coords => {
+        return (
+          parcels.some(
+            coords2 => coords.x === coords2.x && coords.y === coords2.y
+          ) &&
+          !alreadyTraveled.some(
+            coords2 => coords.x === coords2.x && coords.y === coords2.y
+          )
+        )
+      }
+    )
+
+    return this.areConnected(
+      parcels,
+      [...alreadyTraveled, ...neighbours],
+      remaining
+    )
+  }
+
+  handleParcelClick = wallet => (x, y) => {
+    if (!isOwner(wallet, x, y)) {
+      return
+    }
+
+    if (!this.hasNeighbour(x, y)) {
+      return
+    }
```

Comment:
Maybe we can merge these two `if` into one

```
if (!isOwner(wallet, x, y) || !this.hasNeighbour(x, y)) {
       return false
}
```

Also, I would prefer to return `false` instead of `undefined`

Reply:
I don't know if that's the best way for making function validations, the thing is that if we change the 2 ifs statements to one it will execute both of the validations in vain, because if the first one already make the function to not execute anything is a little bit weird to make another validation. What's your opinion on this?

Reply:
I agree to merge the function validations with and OR, in some point maybe a very big or concatenated chain may be difficult to read, but in this case there are only two.

According to the boolean return, why do you prefer returning false instead of nothing? The purpose of this return statement is to stop executing code because is conditional code, also I think that returning a boolean flag will make harder to understand because coders may think that is meant to be use as a function for validating instead of execution.

Reply:
IMHO, it is mostly for consistency and code design. If you need to call this function later and wait for a response, you already have a value (`false`) instead of `undefined`.

Reply:
The thing is that the function must not be called at all for waiting a response, and changing the function signature from void to boolean is going to be a patch for bad usage cases. Although we are not explicitly defining the signature of the function, it exists and we have to respect it. Unless we need to logical change the signature to return a boolean value for a concrete situation, I stand to keep returning anything, because isn't a return, is used as `break`

Reply:
didn't know about that guidelines